ccl/sqlproxyccl: add connection migration-related metrics

jaylim-crl opened the pull request:
#### ccl/sqlproxyccl: add metric that measures connection migration latency

Previously, we added support for connection migration in https://github.com/cockroachdb/cockroach/pull/76805. This commit
adds a new `proxy.conn_migration.attempted.latency` metric that tracks latency
for attempted connection migrations. Having this metric would be beneficial
as we can now know how long users were blocked during connection migrations.

Release justification: Low-risk metric-only change within sqlproxy.

Release note: None

#### ccl/sqlproxyccl: add metric that records the transfer response message size

Informs #76000, and follow up to #76805.

This commit adds a new proxy.conn_migration.transfer_response.message_size
metric that will track the distribution of the transfer response message size.
This will be used to tune a value for the SQL cluster setting:
sql.session_transfer.max_session_size.

Release justification: Low-risk metric-only change within sqlproxy.

Release note: None

jaylim-crl committed:
ccl/sqlproxyccl: add metric that measures connection migration latency

Previously, we added support for connection migration in #76805. This commit
adds a new `proxy.conn_migration.attempted.latency` metric that tracks latency
for attempted connection migrations. Having this metric would be beneficial
as we can now know how long users were blocked during connection migrations.

Release justification: Low-risk metric-only change within sqlproxy.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/77700)
<!-- Reviewable:end -->

jaylim-crl committed:
ccl/sqlproxyccl: add metric that records the transfer response message size

Informs #76000, and follow up to #76805.

This commit adds a new proxy.conn_migration.transfer_response.message_size
metric that will track the distribution of the transfer response message size.
This will be used to tune a value for the SQL cluster setting:
sql.session_transfer.max_session_size.

Release justification: Low-risk metric-only change within sqlproxy.

Release note: None

jeffswenson reviewed (approved):
LGTM

jaylim-crl reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/77700#-:-My7lKZ36NH0stVlecTL:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @JeffSwenson)

___
*[pkg/ccl/sqlproxyccl/conn_migration.go, line 140 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/77700#-MxuZJ2mJqldNAq_iSxV-r2-140:-My7lLaG8Apb-w-zxzG-:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ba4c58e75cc36c17261e3aa66f24170c1b8a3907/pkg/ccl/sqlproxyccl/conn_migration.go#L140)):*
<details><summary><i>Previously, JeffSwenson (Jeff Swenson) wrote…</i></summary><blockquote>

It would be nice to add the transfer latency to the log statements in the defer. That way we can dig into the source of unexpectedly high/low latency.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->
